drizzle-orm/src/singlestore-core/columns/blob.ts

```diff
@@ -0,0 +1,187 @@
+		}
+
+		// for sqlite durable objects
+		// eslint-disable-next-line no-instanceof/no-instanceof
+		if (value instanceof ArrayBuffer) {
+			const decoder = new TextDecoder();
+			return JSON.parse(decoder.decode(value));
+		}
+
+		return JSON.parse(String.fromCodePoint(...value));
+	}
+
+	override mapToDriverValue(value: T['data']): string {
+		return JSON.stringify(value);
+	}
+}
+
+export type SingleStoreBlobBufferBuilderInitial<TName extends string> = SingleStoreBlobBufferBuilder<{
+	name: TName;
+	dataType: 'buffer';
+	columnType: 'SingleStoreBlobBuffer';
+	data: Buffer;
+	driverParam: Buffer;
+	enumValues: undefined;
+}>;
+
+export class SingleStoreBlobBufferBuilder<T extends ColumnBuilderBaseConfig<'buffer', 'SingleStoreBlobBuffer'>>
+	extends SingleStoreColumnBuilder<T>
+{
+	static override readonly [entityKind]: string = 'SingleStoreBlobBufferBuilder';
+
+	constructor(name: T['name']) {
+		super(name, 'buffer', 'SingleStoreBlobBuffer');
+	}
+
+	/** @internal */
+	override build<TTableName extends string>(
+		table: AnySingleStoreTable<{ name: TTableName }>,
+	): SingleStoreBlobBuffer<MakeColumnConfig<T, TTableName>> {
+		return new SingleStoreBlobBuffer<MakeColumnConfig<T, TTableName>>(
+			table,
+			this.config as ColumnBuilderRuntimeConfig<any>,
+		);
+	}
+}
+
+export class SingleStoreBlobBuffer<T extends ColumnBaseConfig<'buffer', 'SingleStoreBlobBuffer'>>
+	extends SingleStoreColumn<T>
+{
+	static override readonly [entityKind]: string = 'SingleStoreBlobBuffer';
+
+	getSQLType(): string {
+		return 'blob';
+	}
+}
+
+export interface BlobConfig<TMode extends BlobMode = BlobMode> {
+	mode: TMode;
+}
+
+/**
+ *  It's recommended to use `text('...', { mode: 'json' })` instead of `blob` in JSON mode, because it supports JSON functions:
+ * >All JSON functions currently throw an error if any of their arguments are BLOBs because BLOBs are reserved for a future enhancement in which BLOBs will store the binary encoding for JSON.
+ *
+ * https://www.sqlite.org/json1.html
+ */
+export function blob(): SingleStoreBlobJsonBuilderInitial<''>;
+export function blob<TMode extends BlobMode = BlobMode>(
+	config?: BlobConfig<TMode>,
+): Equal<TMode, 'bigint'> extends true ? SingleStoreBigIntBuilderInitial<''>
+	: Equal<TMode, 'buffer'> extends true ? SingleStoreBlobBufferBuilderInitial<''>
+	: SingleStoreBlobJsonBuilderInitial<''>;
+export function blob<TName extends string, TMode extends BlobMode = BlobMode>(
+	name: TName,
+	config?: BlobConfig<TMode>,
+): Equal<TMode, 'bigint'> extends true ? SingleStoreBigIntBuilderInitial<TName>
+	: Equal<TMode, 'buffer'> extends true ? SingleStoreBlobBufferBuilderInitial<TName>
+	: SingleStoreBlobJsonBuilderInitial<TName>;
+export function blob(a?: string | BlobConfig, b?: BlobConfig) {
+	const { name, config } = getColumnNameAndConfig<BlobConfig | undefined>(a, b);
+	if (config?.mode === 'json') {
+		return new SingleStoreBlobJsonBuilder(name);
+	}
+	if (config?.mode === 'bigint') {
```

Comment:
Why do we need BigInt mode for blobs? We should support string and maybe buffer modes, but we can start by just mapping string data